File: src/main/java/me/pcasaes/hexoids/core/domain/periodictasks/StalledPlayersPeriodTask.java
package me.pcasaes.hexoids.core.domain.periodictasks;

import me.pcasaes.hexoids.core.domain.eventqueue.GameQueue;
import me.pcasaes.hexoids.core.domain.model.Game;
import me.pcasaes.hexoids.core.domain.model.Player;

import java.util.concurrent.TimeUnit;

public class StalledPlayersPeriodTask implements GamePeriodicTask {

    private final GameQueue gameQueue;

    private StalledPlayersPeriodTask(GameQueue gameQueue) {
        this.gameQueue = gameQueue;
    }

    public static StalledPlayersPeriodTask create(GameQueue gameQueue) {
        return new StalledPlayersPeriodTask(gameQueue);
    }


    @Override
    public long getPeriod() {
        return 60;
    }

    @Override
    public void run() {
        gameQueue.enqueue(() -> Game.get().getPlayers()
                .forEach(Player::expungeIfStalled));
    }

    @Override
    public long getDelay() {
        return 0;
    }

    @Override
    public TimeUnit getTimeUnit() {
        return TimeUnit.SECONDS;
    }
}

File: src/main/kotlin/me/pcasaes/hexoids/core/domain/periodictasks/StalledPlayersPeriodTask.kt
package me.pcasaes.hexoids.core.domain.periodictasks

import me.pcasaes.hexoids.core.domain.eventqueue.GameQueue
import me.pcasaes.hexoids.core.domain.model.Game.Companion.get
import java.util.concurrent.TimeUnit
import java.util.function.Consumer

class StalledPlayersPeriodTask private constructor(private val gameQueue: GameQueue) : GamePeriodicTask {
    override fun getPeriod(): Long {
        return 60
    }

    override fun run() {
        gameQueue.enqueue(Runnable {
            get().getPlayers()
                .forEach(Consumer { obj -> obj.expungeIfStalled() })
        })
    }

    override fun getDelay(): Long {
        return 0
    }

    override fun getTimeUnit(): TimeUnit {
        return TimeUnit.SECONDS
    }

    companion object {
        fun create(gameQueue: GameQueue): StalledPlayersPeriodTask {
            return StalledPlayersPeriodTask(gameQueue)
        }
    }
}
